lib/fetch/body.js

```diff
@@ -397,7 +398,47 @@ function bodyMixinMethods (instance) {
 
       // If mimeType’s essence is "multipart/form-data", then:
       if (/multipart\/form-data/.test(contentType)) {
-        throw new NotSupportedError('multipart/form-data not supported')
+        const headers = {}
+        for (const [key, value] of this.headers) headers[key.toLowerCase()] = value
+
+        const responseFormData = new FormData()
+
+        let busboy
+
+        try {
+          busboy = Busboy({ headers })
+        } catch(err) {
+          // Error due to headers:
+          throw Object.assign(new TypeError(), { cause: err })
+        }
+
+        busboy.on('field', (name, value) => {
+          responseFormData.append(name, value)
+        })
+        busboy.on('file', (name, value, info) => {
+          const { filename, encoding, mimeType } = info
+          const base64 = encoding.toLowerCase() === 'base64'
+          const chunks = []
+          value.on('data', (chunk) => {
+            if (base64) chunk = Buffer.from(chunk.toString(), 'base64')
+            chunks.push(chunk)
+          })
+          value.on('end', () => {
+            const file = new File(chunks, filename, { type: mimeType })
+            responseFormData.append(name, file)
+          })
+        })
+        busboy.on('error', (err) => {
+          throw Object.assign(new TypeError(), { cause: err })
```

Comment:
You can't just throw here... this will crash the process. Is there tests for this?

Reply:
> You can't just throw here... this will crash the process.

Do you have a suggestion?

Reply:
You'll need to reject this promise https://github.com/nodejs/undici/pull/1606/files#diff-bd6db9a8bceb4e9fce3f93ba5284c81f46a44a1187909e8609006e1efeab2570R435